### ISSUE TITLE
test_bad_parsing tests fail with: IOError: [Errno 2] No such file or directory: '/root/ansible_testing/ansible_test_file'

### ISSUE THREAD

**Issue by trinitronx** [ansible/ansible#14789]
##### Issue Type:
- Bug Report
##### Ansible Version:

```
$ ansible --version
ansible 2.1.0 (devel b2a6784a9f) last updated 2016/03/03 23:00:55 (GMT +000)
  lib/ansible/modules/core: (detached HEAD 0bbb7ba38d) last updated 2016/03/03 23:00:59 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD 39e4040685) last updated 2016/03/03 23:01:00 (GMT +000)
  config file =
  configured module search path = /opt/ansible/ansible/library
```
##### Ansible Configuration:
- `N/A`: No `ansible.cfg` / No settings changed from default
- `source`d included ENV script from: `/opt/ansible/ansible/hacking/env-setup`
- `ANSIBLE_HOME=/opt/ansible/ansible`
- `PYTHONPATH=/opt/ansible/ansible/lib:`
##### Environment:

OS running Docker build from: [`Ubuntu 14.04.3 LTS`](https://travis-ci.org/trinitronx/ansible-docker-base/jobs/113524091#L14-L18)
OS running Ansible from: `CentOS Linux 7` (from: Docker image: [centos:7](https://hub.docker.com/_/centos/))
OS managing: `N/A`
##### Summary:

Integration tests for `parsing` make target (playbook: `/opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:29`) fail on current [ansible/ansible@devel](https://github.com/ansible/ansible/tree/b2a6784a9f/test) branch.  Not sure whether to trust that this Ansible build is ok for production use, so I'm blocked on updating my [built container](https://hub.docker.com/r/trinitronx/ansible-base/) to benefit from fixes since it was last built.
##### Steps To Reproduce:
1. Install [ansible/ansible@b2a6784a9f](https://github.com/ansible/ansible/tree/b2a6784a9f/test) to `/opt/ansible/ansible` on a test host
2. Run the included `hacking/env-setup` script: `source /opt/ansible/ansible/hacking/env-setup`
3. Run the included integration tests with `cd /opt/ansible/ansible && make -j 1 parsing test_var_precedence test_templating_settings includes test_hash test_handlers test_group_by`
4. `test_bad_parsing` fails at `/opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:29`

Error looks like:

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501/file", line 2491, in <module>
    main()
  File "/root/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501/file", line 409, in main
    open(path, 'w').close()
IOError: [Errno 2] No such file or directory: '/root/ansible_testing/ansible_test_file'

fatal: [testhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "file"}, "parsed": false}
```
##### Expected Results:

Tests should pass, [output](https://travis-ci.org/trinitronx/ansible-docker-base/jobs/110899113#L3792-L3822) should look like:

```
TASK [test_bad_parsing : file] *************************************************
task path: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:29
ESTABLISH LOCAL CONNECTION FOR USER: root
127.0.0.1 EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1456130434.04-39704483789163 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1456130434.04-39704483789163 `" )'
127.0.0.1 PUT /tmp/tmp6K5guK TO /root/.ansible/tmp/ansible-tmp-1456130434.04-39704483789163/file
127.0.0.1 EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1456130434.04-39704483789163/file; rm -rf "/root/.ansible/tmp/ansible-tmp-1456130434.04-39704483789163/" > /dev/null 2>&1'
changed: [testhost] => {"changed": true, "dest": "/root/ansible_testing/ansible_test_file", "diff": {"after": {"path": "/root/ansible_testing/ansible_test_file", "state": "touch"}, "before": {"path": "/root/ansible_testing/ansible_test_file", "state": "absent"}}, "gid": 0, "group": "root", "invocation": {"module_args": {"backup": null, "content": null, "delimiter": null, "diff_peek": null, "directory_mode": null, "follow": false, "force": false, "group": null, "mode": null, "name": "~/ansible_testing/ansible_test_file", "original_basename": null, "owner": null, "path": "/root/ansible_testing/ansible_test_file", "recurse": false, "regexp": null, "remote_src": null, "selevel": null, "serole": null, "setype": null, "seuser": null, "src": null, "state": "touch", "validate": null}, "module_name": "file"}, "mode": "0644", "owner": "root", "size": 0, "state": "file", "uid": 0}

TASK [test_bad_parsing : include test that we cannot insert arguments] *********
task path: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:32
included: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/scenario1.yml for testhost

TASK [test_bad_parsing : include test that we cannot duplicate arguments] ******
task path: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:36
included: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/scenario2.yml for testhost

TASK [test_bad_parsing : include test that we can't do this for the shell module] ***
task path: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:40
included: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/scenario3.yml for testhost

TASK [test_bad_parsing : include test that we can't go all Little Bobby Droptables on a quoted var to add more] ***
task path: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:44
included: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/scenario4.yml for testhost

TASK [test_bad_parsing : test that a missing/malformed jinja2 filter fails] ****
task path: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:48
fatal: [testhost]: FAILED! => {"failed": true, "msg": "template error while templating string: no filter named 'badfiltername'. String: {{output_dir|badfiltername}}"}
...ignoring

PLAY RECAP *********************************************************************
testhost                   : ok=9    changed=2    unreachable=0    failed=0
```
##### Actual Results:

[Build + integration tests](https://travis-ci.org/trinitronx/ansible-docker-base/jobs/113524091#L3801-L3813) failed with error:

```
TASK [test_bad_parsing : file] *************************************************
task path: /opt/ansible/ansible/test/integration/roles/test_bad_parsing/tasks/main.yml:29
ESTABLISH LOCAL CONNECTION FOR USER: root
127.0.0.1 EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501 `" )'
127.0.0.1 PUT /tmp/tmpXbN09c TO /root/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501/file
127.0.0.1 EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501/file; rm -rf "/root/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501/" > /dev/null 2>&1'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501/file", line 2491, in <module>
    main()
  File "/root/.ansible/tmp/ansible-tmp-1457046096.64-209878160685501/file", line 409, in main
    open(path, 'w').close()
IOError: [Errno 2] No such file or directory: '/root/ansible_testing/ansible_test_file'

fatal: [testhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "file"}, "parsed": false}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @bad_parsing.retry

PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=1

make: *** [parsing] Error 2
```

**Comment by sivel** [comment]
I'm not at a computer right now to see why this could be failing for you, however it is correctly passing on both centos7 and ubuntu1404 on our Travis integration tests that utilize docker containers

**Comment by trinitronx** [comment]
@sivel : Just ran another build today from bb23304b10, and it's [still failing with same error](https://travis-ci.org/trinitronx/ansible-docker-base/jobs/114380321#L3813).  Definitely seems reproducible from my Travis build of forked [ansible-docker-base](https://github.com/trinitronx/ansible-docker-base) if you need a starting point for tracking it down.

```
$ ansible --version
ansible 2.1.0 (devel bb23304b10) last updated 2016/03/07 22:40:45 (GMT +000)
  lib/ansible/modules/core: (detached HEAD 45745424f7) last updated 2016/03/07 22:40:48 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD b51efc51bc) last updated 2016/03/07 22:40:50 (GMT +000)
  config file = 
  configured module search path = /opt/ansible/ansible/library
```

**Comment by sivel** [comment]
Ok, so the problem is in how you are running the tests:

```
docker exec -ti $DOCKER_CONTAINER_ID /bin/sh -c 'cd /opt/ansible/ansible/test/integration/ &&
   make -j 1 parsing test_var_precedence test_templating_settings includes test_hash test_handlers test_group_by '
```

That command starts by running the `parsing` make target.  Unfortunately the `parsing` target has a dependency on the `setup` target having been run.

I have however made `setup` an explicit dependency in the Makefile for `parsing` now (https://github.com/ansible/ansible/commit/8aee648bc98e7e630b815df9dffc64feb9afe632)

**Comment by trinitronx** [comment]
@sivel :  Thanks for the quick response!  Good catch on the make target dependency issue.   FWIW, the only reason I'm running those make targets explicitly is because the ansible build historically never passed _all_ of the make target tests for me.  Since I needed something that "just works", but also wanted to ensure I don't publish docker image builds from `devel` that were failing the most important set of integration tests... otherwise the image we run our playbooks from might end up being broken in fundamental ways.

I'll kick off another build from `devel` now to test out the new `Makefile` change!

Thanks again!

**Comment by trinitronx** [comment]
@sivel :  Well, the good news is that the integration tests now pass!   However, the bad news is that the `nose` unit tests from `make tests` [are now failing](https://travis-ci.org/trinitronx/ansible-docker-base/jobs/114396212#L3779-L3795) :disappointed: ... D'oh!

So, we can consider this issue closed, but now there's another issue with the unit tests on current `devel`

I'll revisit this tomorrow to see what's going on.

**Comment by sivel** [comment]
We are aware of the PlayIterator error, and should have that resolved soonish.

**Comment by trinitronx** [comment]
@sivel : Was able to get a successful build today!  Thanks!
